### DISCUSSION TITLE
Does agent keep track of intercepted requests?

### DISCUSSION THREAD

**Discussion by jethrolam** [nodejs/undici#2163] [Q&A]
In my unit tests, I want to not only intercept the incoming fetch response but also verify the body of any outgoing (intercepted) fetch request.  I searched through the doc but cannot find anything.  Does such a feature exist?

**Comment by arontsang** [comment]
There is an experimental API called interceptor. Its very clunky.
An example of which can be seen in the 3XX redirect handler.
